sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -3,479 +3,622 @@
+  sessionId?: string;
+  timeToLive?: number;
+  to?: string;
+  userProperties?: {
+    [key: string]: any;
+  };
+  viaPartitionKey?: string;
 }
 
 // @public
 export class Sender {
-    cancelScheduledMessage(sequenceNumber: Long): Promise<void>;
-    cancelScheduledMessages(sequenceNumbers: Long[]): Promise<void>;
-    close(): Promise<void>;
-    get isClosed(): boolean;
-    scheduleMessage(scheduledEnqueueTimeUtc: Date, message: SendableMessageInfo): Promise<Long>;
-    scheduleMessages(scheduledEnqueueTimeUtc: Date, messages: SendableMessageInfo[]): Promise<Long[]>;
-    send(message: SendableMessageInfo): Promise<void>;
-    sendBatch(messages: SendableMessageInfo[]): Promise<void>;
-    }
+  cancelScheduledMessage(sequenceNumber: Long, options?: OperationOptions): Promise<void>;
+  cancelScheduledMessages(sequenceNumbers: Long[], options?: OperationOptions): Promise<void>;
+  close(): Promise<void>;
+  get isClosed(): boolean;
+  scheduleMessage(
+    scheduledEnqueueTimeUtc: Date,
+    message: SendableMessageInfo,
+    options?: OperationOptions
+  ): Promise<Long>;
+  scheduleMessages(
+    scheduledEnqueueTimeUtc: Date,
+    messages: SendableMessageInfo[],
+    options?: OperationOptions
+  ): Promise<Long[]>;
+  send(message: SendableMessageInfo, options?: OperationOptions): Promise<void>;
+  sendBatch(messages: SendableMessageInfo[], options?: OperationOptions): Promise<void>;
+}
 
-// @public
+// @public (undocumented)
 export class ServiceBusClient {
-    constructor(connectionString: string, options?: ServiceBusClientOptions);
-    constructor(host: string, credential: TokenCredential, options?: ServiceBusClientOptions);
-    close(): Promise<any>;
-    createQueueClient(queueName: string): QueueClient;
-    createSubscriptionClient(topicName: string, subscriptionName: string): SubscriptionClient;
-    createTopicClient(topicName: string): TopicClient;
-    readonly name: string;
+  constructor(connectionString: string, options?: ServiceBusClientOptions);
+  constructor(
+    hostName: string,
+    tokenCredential: TokenCredential_2,
+    options?: ServiceBusClientOptions
+  );
+  // (undocumented)
+  close(): Promise<void>;
+  createReceiver(queueName: string, receiveMode: "peekLock"): NonSessionReceiver<"peekLock">;
+  createReceiver(
+    queueName: string,
+    receiveMode: "receiveAndDelete"
+  ): NonSessionReceiver<"receiveAndDelete">;
+  createReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "peekLock"
+  ): NonSessionReceiver<"peekLock"> & SubscriptionRuleManagement;
+  createReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "receiveAndDelete"
+  ): NonSessionReceiver<"receiveAndDelete"> & SubscriptionRuleManagement;
+  createSender(queueOrTopicName: string): Sender;
+  createSessionReceiver(
+    queueName: string,
+    receiveMode: "peekLock",
+    sessionId: string | "",
+    options?: CreateSessionReceiverOptions
+  ): SessionReceiver<"peekLock">;
```

Comment:
From the service team guidance it sounds like we want this method to _actually_ obtain the session lock and not wait until the user first does something (specifically when the user does not specify a session ID)

If that's the case this method should be changed to return a Promise.